validation/src/shared_table/mod.rs

```diff
@@ -519,14 +520,27 @@ impl SharedTable {
 		// aggregation in the future.
 		let table_attestations = self.inner.lock().table.proposed_candidates(&*self.context);
 		table_attestations.into_iter()
-			.map(|attested| AttestedCandidate {
-				candidate: attested.candidate,
-				validity_votes: attested.validity_votes.into_iter().map(|(a, v)| match v {
-					GAttestation::Implicit(s) => (a, ValidityAttestation::Implicit(s)),
-					GAttestation::Explicit(s) => (a, ValidityAttestation::Explicit(s)),
-				}).collect(),
-			})
-			.collect()
+			.map(|attested| {
+				let mut validator_indices = BitVec::new();
+				let validity_votes = attested.validity_votes.into_iter().map(|(a, v)| {
+					let a = a as usize;
+					if validator_indices.len() <= a {
+						validator_indices.resize(a + 1, false);
+					}
+					validator_indices.set(a, true);
+
+					match v {
+						GAttestation::Implicit(s) => ValidityAttestation::Implicit(s),
+						GAttestation::Explicit(s) => ValidityAttestation::Explicit(s),
+					}
+				}).collect();
+
+				AttestedCandidate {
+					candidate: attested.candidate,
+					validity_votes,
```

Comment:
are these guaranteed to be sorted?

Reply:
the runtime expects the i'th `true` bit to be paired with the i'th member of this list. but there is nothing here indicating to me that guarantees that.